### PR TITLE
[SOAR-285] Update Grep plugin to fix security issues and file creation/deletion issues

### PR DESCRIPTION
**Pull request by mhofert-r7**
### Description

Describe the proposed changes:

  - Update Grep plugin to fix security issues and file creation/deletion issues

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "rapid7/Grep:1.0.4. Step name: match_base64\n",
    "meta": {},
    "output": {
      "found": true,
      "hits": 2,
      "matches": [
        "I see the dog ",
        " The dog sees me "
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/grep:1.0.4 run < tests/match_base64.json
</summary>
</details>

### Run

<details>

```
{
  "body": {
    "log": "rapid7/Grep:1.0.4. Step name: match_string\n",
    "meta": {},
    "output": {
      "found": true,
      "hits": 1,
      "matches": [
        "I see the dog"
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/grep:1.0.4 run < tests/match_string.json
</summary>
</details>

### CODE REVIEW

**Review by jmcadams-r7 — DISMISSED** [review]

File: grep/icon_grep/util/utils.py [hunk 1]
```diff
@@ -1,58 +1,33 @@
-import komand
+from insightconnect_plugin_runtime.exceptions import PluginException
+from subprocess import run, CalledProcessError
```
Comment (mrinehart-r7) [inline grep/icon_grep/util/utils.py:2]:
```suggestion
from subprocess import run
```
I don't see CalledProcessError used anywhere

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: grep/icon_grep/util/utils.py [hunk 2]
```diff
@@ -1,58 +1,33 @@
...
-grep = "egrep"
-wc = "wc"
 
-FOUND = "found"
-HITS = "hits"
-MATCHES = "matches"
-TEMP_PATH = tempfile.mkdtemp() + "/"
+def run_grep(log: object, text: str, pattern: str, behavior: str) -> str:
+    with tempfile.NamedTemporaryFile() as fp:
+        fp.write(text.encode())
```
Comment (rnanopoulos-r7) [inline grep/icon_grep/util/utils.py:8]: Do we need to impose any limit on the file size here? What will happen if the data is more than can be stored temporarily in the plugin container?

**Review by jmcadams-r7 — APPROVED** [review]

**Review by ncramer-r7 — APPROVED** [review]

File: grep/icon_grep/util/utils.py (hunk 2 above)
Comment (jschipp-r7) [reply grep/icon_grep/util/utils.py:9]: This something we should think about for sure.

File: grep/help.md [hunk 3]
```diff
@@ -113,6 +133,7 @@ _This plugin does not contain any troubleshooting information._
 
 # Version History
 
+* 1.0.4 - Security fix for potential arbitrary code execution | Fix issue where grep may not find file to run against
```
Comment (jschipp-r7) [inline grep/help.md:136]:
```suggestion
* 1.0.4 - Security fix for potential arbitrary code execution | Fix issue where Grep may not find file to run against
```

**Review by jschipp-r7 — APPROVED** [review]

**Pull request merged**